data-raw/simulate-data.R

```diff
@@ -0,0 +1,240 @@
+    purrr::reduce(\(integer1, integer2) paste(integer1, integer2, sep = "")) |>
+    pad_integers(width = length)
+}
+
+create_fake_npu <- function(n) {
+  stringr::str_c(
+    "NPU",
+    create_padded_integer(n, 8)
+  )
+}
+
+create_fake_hovedspeciale_ans <- function(n) {
+  "https://www.dst.dk/da/Statistik/dokumentation/Times/forebyggelsesregistret/spec" |>
+    read_html() |>
+    html_element("table") |>
+    html_table() |>
+    pull(Tekst) |>
+    sample(n, replace = TRUE)
+}
+
+create_fake_drug_name <- function(atc) {
+  codeCollection::ATCKoodit |>
+    tibble::as_tibble() |>
+    dplyr::filter(Koodi %in% atc) |>
+    dplyr::pull(en) |>
+    sample(length(atc), replace = TRUE)
+}
+
+to_wwyy <- function(x) {
+  format(lubridate::as_date(x), format = "%W%y")
+}
+
+to_yyyymmdd <- function(x) {
+  format(lubridate::as_date(x), format = "%Y%m%d")
+}
+
+# Insert extra values ------------------------------------------------------
+
+insertion_rate <- function(proportion) {
+  runif(1) < proportion
+}
+
+insert_specific_atc <- function(data, proportion = 0.3) {
+  # c(
+  #       "A10AB",
+  #       "A10AC",
+  #       "A10AD",
+  #       "A10AE",
+  #       "A10BA",
+  #       "A10BB",
+  #       "A10BD",
+  #       "A10BG",
+  #       "A10BH",
+  #       "A10BJ",
+  #       "A10BK",
+  #       "A10BX"
+  # )
+  data |>
+    mutate(
+      across(
+        matches("^atc$"),
+        \(column) if_else(
+          runif(n()) < proportion,
+          sample(c("A10BA02", "A10BJ02", "A10BJ06"), 1),
+          column
+        )
+      )
+    )
+}
+
+# Insert a few false-positive cases with purchases of metformin:
+insert_false_metformin <- function(data, proportion = 0.05) {
+  if (!all(colnames(data) %in% c("atc", "name"))) {
+    return(data)
+  }
+  data |>
+    dplyr::mutate(
+      atc = dplyr::if_else(
+        indo %in% c("0000092", "0000276", "0000781") & insertion_rate(proportion),
+        "A10BA02",
+        atc
+      ),
+      name = dplyr::if_else(
+        indo %in% c("0000092", "0000276", "0000781") & insertion_rate(proportion),
+        "metformin",
+        name
+      )
+    )
+}
+
+# Insert some false positives for Wegovy and Saxenda.
+insert_false_drug_names <- function(data, proportion = 0.05) {
+  if (!all(colnames(data) %in% c("atc", "name"))) {
+    return(data)
+  }
+  data |>
+    mutate(
+      name = case_when(
+        atc == "A10BJ06" & insertion_rate(proportion) ~ "Wegovy Flextouch",
+        atc == "A10BJ02" & insertion_rate(proportion) ~ "Saxenda",
+        TRUE ~ name
+      )
+    )
+}
+
+insert_analysiscode <- function(x) {
+  # TODO: Is this necessary?
+  # ANALYSISCODE: npu code of analysis type (chr)
+  # 50% is either NPU27300 or NPU03835
+  # other 50% is NPU10000 to NPU99999
+  x
+}
+
+# Generate data -----------------------------------------------------------
+
+# use the simulation definition data to simulate some data
+simulate_data <- function(data, n) {
+  N <- n
```

Comment:
Not sure why this is necessary? :)

Reply:
This is a `fabricator` package thing, since the N is used to set the sample size for simulated data created by fabricator. I'll add a comment though.